src/utils/transaction-guards.ts

```diff
@@ -412,3 +412,19 @@ export const isERC20Transfer = (value: TransferInfo): value is Erc20Transfer =>
 export const isERC721Transfer = (value: TransferInfo): value is Erc721Transfer => {
   return value.type === TransactionTokenType.ERC721
 }
+
+/**
+ * True if the tx calls `approveHash`
+ */
+export const isOnChainConfirmationTxData = (data?: TransactionData): boolean => {
+  const approveHashSelector = Safe__factory.createInterface().getFunction('approveHash').selector
+
+  return Boolean(data && data.hexData?.startsWith(approveHashSelector))
```

Comment:
Nit: we don't need to create the interface each time.
```suggestion
const approveHashSelector = Safe__factory.createInterface().getFunction('approveHash').selector
export const isOnChainConfirmationTxData = (data?: TransactionData): boolean => {
  return Boolean(data && data.hexData?.startsWith(approveHashSelector))
```